src/network/MissionControlProtocol.cpp

```diff
@@ -76,10 +76,73 @@ void MissionControlProtocol::handleOperationModeRequest(const json& j) {
 	}
 }
 
+static bool validateDriveModeRequest(const json& j) {
+	return util::validateKey(j, "mode", val_t::string) &&
+		   util::validateOneOf(j, "mode", {"normal", "turn-in-place", "crab"});
+}
+
+void MissionControlProtocol::handleDriveModeRequest(const json& j) {
+	std::string mode = j["mode"];
+	if (mode == "normal") {
+		Globals::driveMode = DriveMode::Normal;
+		for (int i = 1; i <= 4; i++) {
+			robot::setMotorPos(Constants::WHEEL_IDS[i], Constants::NORMAL_WHEEL_ROTS[i]);
+		}
+	} else if (mode == "turn-in-place") {
+		Globals::driveMode = DriveMode::TurnInPlace;
+		for (int i = 1; i <= 4; i++) {
+			robot::setMotorPos(Constants::WHEEL_IDS[i],
+							   Constants::TURN_IN_PLACE_WHEEL_ROTS[i]);
+		}
+	} else if (mode == "crab") {
+		Globals::driveMode = DriveMode::Crab;
+		for (int i = 1; i <= 4; i++) {
+			robot::setMotorPos(Constants::WHEEL_IDS[i], Constants::CRAB_WHEEL_ROTS[i]);
+		}
+	}
+}
+
 static bool validateDriveRequest(const json& j) {
 	std::string msg = j.dump();
-	return util::hasKey(j, "straight") && util::validateRange(j, "straight", -1, 1) &&
-		   util::hasKey(j, "steer") && util::validateRange(j, "steer", -1, 1);
+	if (Globals::driveMode != DriveMode::Normal) {
+		LOG_F(WARNING, "Drive mode set to %s, not Normal",
+			  driveModeStrings.at(Globals::driveMode).c_str());
+	}
+	return Globals::driveMode == DriveMode::Normal && util::hasKey(j, "straight") &&
+		   util::validateRange(j, "straight", -1, 1) && util::hasKey(j, "steer") &&
+		   util::validateRange(j, "steer", -1, 1);
+}
+
+static bool validateTankDriveRequest(const json& j) {
+	std::string msg = j.dump();
```

Comment:
I know we do this in a lot of places, but why do we `dump()` and then don't use `msg`?

Reply:
I actually don't know - it was done in `validateDriveRequest` and so I emulated it in the others. I'll remove them for now and revert if needed